### PR TITLE
feat: Add filter_policy to chroma integration

### PR DESCRIPTION
**Pull request by vblagoje**
### Why:
Adds flexible filtering options to chroma integration. By introducing a filter policy option (`replace` or `merge`), developers can now control how runtime filters are applied relative to initialization time filters. 

- partially fixes https://github.com/deepset-ai/haystack-core-integrations/issues/780
- supersedes https://github.com/deepset-ai/haystack-core-integrations/pull/781

### What:
- Added `filter_policy` parameter with options `replace` and `merge` across multiple retrievers to control filter behavior dynamically.
- Unit tests were updated or added to validate the new functionality.

### How can it be used:

- **Dynamic Filter Behavior Adjustment:**  Users can decide whether to completely override the initial filters set during the retriever's initialization (`replace`) or merge them with runtime filters, with the latter taking precedence (`merge`).

- **Complex Search Scenarios:**  
    - In cases where the context of a query might dictate altering pre-set filters without discarding them, the `merge` option allows for an additive approach.
    - For strict query contexts that require ignoring initial filters, the `replace` option offers a clean slate for filters at runtime.

### How did you test it:
- Unit tests were enhanced or newly created to cover both `replace` and `merge` scenarios for the `filter_policy` parameter. 
- Tests ensure that filter logic is correctly applied based on the policy setting, whether it merges runtime filters with initial filters or replaces them entirely.
- Additional test cases should be considered for complex filter merge scenarios to ensure priority and override mechanisms function as expected.

### CODE REVIEW

**Comment by vblagoje** [comment]
Ready for review now @julian-risch 

- Since you last looked at this PR we introduced `apply_filter_policy` function that is used by all Retrievers in this [PR](https://github.com/deepset-ai/haystack/pull/7902) and we thoroughly tested that function in that PR. We then just applied it here in run method
- Please note if FilterPolicy type is used consistently both in init and in run, and that it is documented well

**Comment by vblagoje** [comment]
Stefano recommended one more small improvement. Please hold your review until a new commit with it is added here

**Comment by vblagoje** [comment]
@julian-risch please have a look at https://github.com/deepset-ai/haystack-core-integrations/pull/827 @anakin87 has already gone through a through review of Astra and should help you validate this PR is well-aligned as well

File: integrations/chroma/src/haystack_integrations/components/retrievers/chroma/retriever.py [hunk 1]
```diff
@@ -120,9 +137,17 @@ def run(
         Run the retriever on the given input data.
 
         :param query_embedding: the query embeddings.
+        :param filters: Filters applied to the retrieved Documents. The way runtime filters are applied depends on
+                        the `filter_policy` chosen at document store initialization. See init method docstring for more
+                        details.
+        :param top_k: the maximum number of documents to retrieve.
```
Comment (julian-risch) [inline integrations/chroma/src/haystack_integrations/components/retrievers/chroma/retriever.py:143]: very good 👍

**Review by julian-risch — APPROVED** [review]
Code changes look good to me! Similar to me comment on the qdrant PR: I suggest updating the general page here: https://docs.haystack.deepset.ai/docs/retrievers
And we could consider including it in https://docs.haystack.deepset.ai/docs/chromaqueryretriever but I don't think it's required.
Cc @dfokina

**Pull request merged**